pkg/api/controller.go

```diff
@@ -5178,13 +5181,50 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
 }
 
 func (c *Controller) ListPullRequests(w http.ResponseWriter, r *http.Request, repository string, params apigen.ListPullRequestsParams) {
-	writeResponse(w, r, http.StatusNotImplemented, nil)
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ListPullRequestsAction,
+			Resource: permissions.RepoArn(repository),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "list_pull_requests", r, repository, "", "")
+
+	res, hasMore, err := c.Catalog.ListPullRequest(ctx, repository, paginationPrefix(params.Prefix), paginationAmount(params.Amount), paginationAfter(params.After), pullRequestStatus(params.Status))
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+
+	pulls := make([]apigen.PullRequest, 0, len(res))
+	for _, p := range res {
+		pulls = append(pulls, apigen.PullRequest{
+			PullRequestBasic: apigen.PullRequestBasic{
+				// Explicitly not passing description so to maintain permissions requirements
+				Description: nil,
```

Comment:
Can't this row simply be removed?

Reply:
It can but we want to make sure the next person understands why it's nil